conjure-java-core/src/main/java/com/palantir/conjure/java/services/dialogue/ClassVisitor.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.services.dialogue;
+
+import com.palantir.conjure.java.types.ClassNameVisitor;
+import com.palantir.conjure.java.types.DefaultClassNameVisitor;
+import com.palantir.conjure.spec.ExternalReference;
+import com.palantir.conjure.spec.ListType;
+import com.palantir.conjure.spec.MapType;
+import com.palantir.conjure.spec.OptionalType;
+import com.palantir.conjure.spec.PrimitiveType;
+import com.palantir.conjure.spec.SetType;
+import com.palantir.conjure.spec.TypeDefinition;
+import com.squareup.javapoet.TypeName;
+import java.io.InputStream;
+import java.util.List;
+
+final class ClassVisitor implements ClassNameVisitor {
+
+    enum Mode {
```

Comment:
Might be worth considering using a boolean instead of an enum is there are only two values and only one we actually use. The other one exists only as the complementary.
`isReturnValue` ?

Reply:
I think it's easier to read as is